src/kernels/execution/helpers.ts

```diff
@@ -353,6 +355,10 @@ export type CellOutputMetadata = {
      * Whether to display the open plot icon.
      */
     __displayOpenPlotIcon?: boolean;
+    /**
+     * A unique identifier to track a specific output.
+     */
+    id?: string;
```

Comment:
Maybe we could call this `__id` which would be similar to `__isJson` and `__displayOpenPlotIcon` so even if in the future the core introduces `CellOutputMetadata.id` we won't have a conflict.